DOCS-7681 / SE-13759 Calling Oracle Stored Procedure with TO_DATE() by DB (#696)
mulesoft/docs-connectors: modules/ROOT/pages/db/database-connector-examples.adoc

For Database Connector version 1.4.0 or later, make sure that no whitespace is between the name of the stored procedure and the first parentheses.

> Database Connector supports the use of callable statements with the format ``{ call procedureName(:param1, :param2, ..., :paramN) }`` where parameters matches positionally, that is to say, `:paramN` matches the `Nth` parameter in the stored procedure declaration.

Reviewer: You have already introduced this as "Anypoint Connector for Database (Database Connector)"?

Reviewer: ```suggestion
Database Connector supports the use of callable statements with the format ``{ call procedureName(:param1, :param2, ..., :paramN) }``, in which the parameter number matches its position in the stored procedure declaration.
```
I don't think that you have to explain paramN. Maybe just end the example after ...,?

> Suppose you are using an Oracle database which you have initialized using with the following table and stored procedure:

Reviewer: Clarify the "using" / "with" wording by choosing one that matches your intent:
Was the database initialized by using the table and procedure?
Or was the database initialized using the values in the table and the procedure?
Or was the database initialized to include the values in the table and the procedure?

[source,xml,linenums]
----
    CREATE TABLE SYSTEM.employees(
        employee_id INTEGER GENERATED BY DEFAULT AS IDENTITY,
        employee_name VARCHAR2(100),
        employee_age INTEGER,
        employee_birthday TIMESTAMP,
        PRIMARY KEY(employee_id)
    );

    CREATE PROCEDURE createEmployee(e_name VARCHAR2, e_age NUMBER, e_birth_date DATE) AS
    BEGIN
        INSERT INTO SYSTEM.employees(employee_name, employee_age, employee_birthday) VALUES(e_name, e_age, e_birth_date);
    END;
----

> The connector does not support named parameters in callable statements, so attempting to match parameters by name ( ``{ call createEmployee(employee_age => :age, e_birth_date => :date, e_name => :name) }`` ) does not work. Parameters must be provided in the appropriate order: `{ call createEmployee(e_name => :name, employee_age => :age, e_birth_date => :date) }`.

Reviewer: ```suggestion
The connector does not support named parameters in callable statements, so attempting to match parameters by name ( ``{ call createEmployee(employee_age => :age, e_birth_date => :date, e_name => :name) }`` ) does not work. Parameters must be provided in the appropriate order: 
+
+ `{ call createEmployee(e_name => :name, employee_age => :age, e_birth_date => :date) }`.
```

> [[use_dates_stored_procedures]]

Reviewer: I have not seen this formatting; is it okay?

== Use Dates on Stored Procedures

> Database Connector does not support the use of engine specific embedded functions, for example, if you want to change a date to a specific format before calling a procedure in an Oracle database, the following approach does not work:  `{ call createEmployee(e_name => :name, employee_age => :age, e_birth_date => TO_DATE(:date, 'YYYY-MM-DD HH:mm:ss')) }.` +

Reviewer: ```suggestion
Database Connector does not support the use of engine-specific embedded functions. For example, if you want to change a date to a specific format before calling a procedure in an Oracle database, the following approach does not work:  
+
+ `{ call createEmployee(e_name => :name, employee_age => :age, e_birth_date => TO_DATE(:date, 'YYYY-MM-DD HH:mm:ss')) }.` +
```

> Use DataWeave for all data transformations before the invocation to the callable statement, for the previous example we perform the following transformation:

Reviewer: ```suggestion
Use DataWeave for all data transformations before the invocation to the callable statement. Using the previous example, for instance, perform the following transformation:
```
